src/db/transaction.rs

```diff
@@ -51,7 +51,7 @@ impl DbTransactionPool {
         let result = match (self.lock_collection.clone(), self.is_read) {
             (Some(lc), true) => db.lock_for_read(lc).await,
             (Some(lc), false) => db.lock_for_write(lc).await,
-            _ => Ok(()),
+            (None, is_read) => db.begin(!is_read).await,
```

Comment:
Nice, this fix is much simpler than I thought it'd be.

One issue though: we only correctly set `is_read` (in the extractor below) when a collection exists. The `/info` endpoints lack a collection and always get `is_read = true` which is good.

However the `delete_all` handler currently assumes nothing calls `begin` for it (which is no longer the case) so it calls `begin(true)` itself. We should fix the `is_read` value for it by always checking the http method whether a collection was found or not, then we can remove `delete_all`'s manual `begin(true)`